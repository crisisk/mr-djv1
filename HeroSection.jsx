import React, { useId } from 'react'
import { useTranslation } from 'react-i18next'
import Button from './Buttons.jsx'

const HeroSection = ({
  title,
  subtitle,
  ctaPrimaryText,
  ctaSecondaryText,
  backgroundClass = 'bg-neutral-dark',
  titleColor = 'text-secondary',
  subtitleColor = 'text-neutral-light',
  children,
}) => {
  const { t } = useTranslation()

  const resolvedTitle = title ?? t('hero.title')
  const resolvedSubtitle = subtitle ?? t('hero.subtitle')
  const resolvedPrimaryCta = ctaPrimaryText ?? t('hero.ctaPrimaryText')
  const resolvedSecondaryCta = ctaSecondaryText ?? t('hero.ctaSecondaryText')

  const ctaGroupLabelId = useId()
  const ctaGroupDescriptionId = useId()
  const statsGroupLabelId = useId()

  return (
    <div
      className={`${backgroundClass} hero-motion py-spacing-3xl px-spacing-xl min-h-[60vh] flex items-center`}
    >
      <div className="container mx-auto text-center hero-motion__inner">
        <h1 className={`text-font-size-h1 font-extrabold ${titleColor} mb-spacing-md hero-motion__headline`}>
          {resolvedTitle}
        </h1>
        <p
          className={`text-font-size-h3 mb-spacing-xl max-w-4xl mx-auto ${subtitleColor} hero-motion__subtitle`}
        >
          {resolvedSubtitle}
        </p>
        <div className="flex justify-center space-x-spacing-md hero-motion__ctaGroup">
          <Button variant="secondary" size="lg" className="hero-motion__cta">
            {resolvedPrimaryCta}
          </Button>
          {resolvedSecondaryCta && (
            <Button variant="outline" size="lg" className="hero-motion__cta hero-motion__cta--secondary">
              {resolvedSecondaryCta}
            </Button>
          )}
        </div>
        {children && <div className="hero-motion__statsSlot">{children}</div>}
      </div>
    </div>
  )
}

export default HeroSection
