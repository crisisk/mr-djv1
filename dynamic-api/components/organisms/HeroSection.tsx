import React, { ReactNode, MouseEvent } from 'react';
import Button from '../ui/Button';

// Define the allowed Button variants and sizes based on requirements
type ButtonVariant = 'primary' | 'secondary' | 'ghost';
type ButtonSize = 'sm' | 'md' | 'lg';

// Define the structure for a metric item
interface Metric {
  label: string;
  value: string;
}

// Define the props for the HeroSection component
interface HeroSectionProps {
  eyebrow?: string;
  title: string;
  subtitle: string;
  ctaPrimaryText?: string;
  ctaSecondaryText?: string;
  ctaPrimaryHref?: string;
  ctaSecondaryHref?: string;
  onPrimaryClick?: (event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
  onSecondaryClick?: (event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
  backgroundClass?: string;
  titleColor?: string;
  subtitleColor?: string;
  badges?: string[];
  socialProof?: string;
  metrics?: Metric[];
  children?: ReactNode;
}

const HeroSection: React.FC<HeroSectionProps> = ({
  eyebrow,
  title,
  subtitle,
  ctaPrimaryText,
  ctaSecondaryText,
  ctaPrimaryHref,
  ctaSecondaryHref,
  onPrimaryClick,
  onSecondaryClick,
  backgroundClass = 'bg-neutral-dark',
  titleColor = 'text-secondary',
  subtitleColor = 'text-neutral-light',
  badges = [],
  socialProof,
  metrics = [],
  children,
}) => {
  // Helper function to render a button, handling both internal clicks and external links
  const renderButton = (
    text: string | undefined,
    href: string | undefined,
    onClick: ((event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void) | undefined,
    variant: ButtonVariant,
  ) => {
    if (!text) return null;

    const handleClick = (e: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => {
      // Prevent default navigation if an onClick handler is provided without an href
      if (!href && onClick) {
        e.preventDefault();
      }
      onClick?.(e);
    };

    const buttonElement = (
      <Button
        variant={variant}
        size="lg" // Use 'lg' size as per original component logic
        onClick={!href ? (handleClick as (e: MouseEvent<HTMLButtonElement>) => void) : undefined}
        // If href is present, the Button is wrapped in <a>, so onClick is handled by <a>
      >
        {text}
      </Button>
    );

    if (href) {
      return (
        <a
          href={href}
          className="inline-flex"
          onClick={handleClick as (e: MouseEvent<HTMLAnchorElement>) => void}
          // Ensure the link is accessible
          role="button"
          aria-label={text}
        >
          {buttonElement}
        </a>
      );
    }

    return buttonElement;
  };

  return (
    <div className={`${backgroundClass} py-spacing-3xl px-spacing-xl min-h-[60vh] flex items-center`}>
      <div className="container mx-auto text-center space-y-spacing-lg">
        {eyebrow && (
          <p className="text-sm uppercase tracking-[0.3em] text-secondary/80" aria-label="Category">
            {eyebrow}
          </p>
        )}
        <h1 className={`text-font-size-h1 font-extrabold ${titleColor}`}>
          {title}
        </h1>
        <p className={`text-font-size-h3 max-w-4xl mx-auto ${subtitleColor}`}>
          {subtitle}
        </p>
        {badges.length > 0 && (
          <div className="flex flex-wrap justify-center gap-spacing-sm" aria-label="Features">
            {badges.map((badge) => (
              <span
                key={badge}
                className="rounded-full bg-neutral-light/10 px-spacing-md py-spacing-xs text-sm font-semibold text-neutral-light"
              >
                {badge}
              </span>
            ))}
          </div>
        )}
        <div className="flex flex-wrap justify-center gap-spacing-md">
          {/* Primary CTA - using 'secondary' variant as per original JSX */}
          {renderButton(ctaPrimaryText, ctaPrimaryHref, onPrimaryClick, 'secondary')}
          {/* Secondary CTA - using 'secondary' variant */}
          {ctaSecondaryText &&
            renderButton(ctaSecondaryText, ctaSecondaryHref, onSecondaryClick, 'secondary')}
        </div>
        {socialProof && (
          <p className="text-sm text-neutral-light/80 max-w-3xl mx-auto" aria-label="Social proof">
            {socialProof}
          </p>
        )}
        {metrics.length > 0 && (
          <dl className="grid gap-spacing-lg md:grid-cols-3 text-left text-neutral-light/90 mt-spacing-xl">
            {metrics.map((metric) => (
              <div key={`${metric.label}-${metric.value}`} className="text-center md:text-left">
                <dt className="text-xs uppercase tracking-wide text-neutral-light/60">
                  {metric.label}
                </dt>
                <dd className="text-font-size-h2 font-bold" aria-label={`${metric.label} value`}>
                  {metric.value}
                </dd>
              </div>
            ))}
          </dl>
        )}
        {children}
      </div>
    </div>
  );
};

export default HeroSection;