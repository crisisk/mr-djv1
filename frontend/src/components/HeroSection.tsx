import type { ReactNode } from 'react';
import { Button } from './Button';

export interface HeroSectionProps {
  title: string;
  subtitle: string;
  ctaPrimaryText: string;
  ctaSecondaryText?: string;
  backgroundClass?: string;
  titleColor?: string;
  subtitleColor?: string;
  children?: ReactNode;
}

export function HeroSection({
  title,
  subtitle,
  ctaPrimaryText,
  ctaSecondaryText,
  backgroundClass = 'bg-neutral-dark',
  titleColor = 'text-secondary',
  subtitleColor = 'text-neutral-light',
  children,
}: HeroSectionProps) {
  return (
    <section className={`${backgroundClass} py-spacing-3xl px-spacing-xl min-h-[60vh] flex items-center`}>
      <div className="container mx-auto text-center space-y-spacing-xl">
        <div>
          <h1 className={`heading-1 ${titleColor} mb-spacing-md`}>{title}</h1>
          <p className={`lead mb-spacing-xl max-w-4xl mx-auto ${subtitleColor}`}>{subtitle}</p>
        </div>
        <div className="flex justify-center gap-spacing-md flex-wrap">
          <Button variant="secondary" size="lg">
            {ctaPrimaryText}
          </Button>
          {ctaSecondaryText ? (
            <Button variant="outline" size="lg">
              {ctaSecondaryText}
            </Button>
          ) : null}
        </div>
        {children}
      </div>
    </section>
  );
}

export default HeroSection;
