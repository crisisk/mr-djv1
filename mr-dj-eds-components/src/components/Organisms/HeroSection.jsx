import React from 'react';
import Button from '../Atoms/Buttons.jsx';

const HeroSection = ({
  title,
  subtitle,
  ctaPrimaryText,
  ctaSecondaryText,
  backgroundClass = 'bg-neutral-dark',
  backgroundImage = null,
  titleColor = 'text-secondary',
  subtitleColor = 'text-neutral-light',
  variant = 'A',
  children,
}) => {
  const bgStyle = backgroundImage ? {
    backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url(${backgroundImage})`,
    backgroundSize: 'cover',
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat'
  } : {};

  // Handle CTA button clicks with GTM tracking
  const handlePrimaryClick = () => {
    if (window.dataLayer) {
      window.dataLayer.push({
        event: 'cta_click',
        cta_type: 'primary',
        cta_text: ctaPrimaryText,
        variant: variant,
        section: 'hero'
      });
    }
  };

  const handleSecondaryClick = () => {
    if (window.dataLayer) {
      window.dataLayer.push({
        event: 'cta_click',
        cta_type: 'secondary',
        cta_text: ctaSecondaryText,
        variant: variant,
        section: 'hero'
      });
    }
  };

  return (
    <div
      className={`${!backgroundImage ? backgroundClass : ''} py-spacing-3xl px-spacing-xl min-h-[60vh] flex items-center relative`}
      style={bgStyle}
    >
      <div className="container mx-auto text-center relative z-10">
        <h1 className={`text-font-size-h1 font-extrabold ${titleColor} mb-spacing-md ${backgroundImage ? 'text-white drop-shadow-lg' : ''}`}>
          {title}
        </h1>
        <p className={`text-font-size-h3 mb-spacing-xl max-w-4xl mx-auto ${subtitleColor} ${backgroundImage ? 'text-white drop-shadow-md' : ''}`}>
          {subtitle}
        </p>
        <div className="flex justify-center space-x-spacing-md">
          <Button variant="secondary" size="lg" onClick={handlePrimaryClick}>
            {ctaPrimaryText}
          </Button>
          {ctaSecondaryText && (
            <Button variant="outline" size="lg" onClick={handleSecondaryClick}>
              {ctaSecondaryText}
            </Button>
          )}
        </div>
        {children}
      </div>
    </div>
  );
};

export default HeroSection;
