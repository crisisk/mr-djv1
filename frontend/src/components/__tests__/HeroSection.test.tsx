import { render } from "@testing-library/react";
import { HeroSection } from "../HeroSection";

describe("HeroSection", () => {
  it("matches the default snapshot", () => {
    const { container } = render(
      <HeroSection
        title="Ervaar de ultieme avond"
        subtitle="Van ceremoniÃ«le momenten tot de laatste dans, wij verzorgen het muzikale hoogtepunt."
        ctaPrimaryText="Controleer beschikbaarheid"
        ctaSecondaryText="Vraag een offerte aan"
      >
        <p>Inclusief licht- en geluidspakket op maat.</p>
      </HeroSection>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
