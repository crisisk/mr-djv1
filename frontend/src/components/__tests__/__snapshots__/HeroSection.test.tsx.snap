// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection matches the default snapshot 1`] = `
<section
  class="bg-neutral-dark py-spacing-3xl px-spacing-xl min-h-[60vh] flex items-center"
>
  <div
    class="container mx-auto text-center space-y-spacing-xl"
  >
    <div>
      <h1
        class="heading-1 text-secondary mb-spacing-md"
      >
        Ervaar de ultieme avond
      </h1>
      <p
        class="lead mb-spacing-xl max-w-4xl mx-auto text-neutral-light"
      >
        Van ceremoniÃ«le momenten tot de laatste dans, wij verzorgen het muzikale hoogtepunt.
      </p>
    </div>
    <div
      class="flex justify-center gap-spacing-md flex-wrap"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-strong transition-colors duration-200 disabled:opacity-60 disabled:cursor-not-allowed focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 bg-secondary text-neutral-dark hover:bg-secondary/90 focus-visible:ring-secondary/40 px-spacing-lg py-spacing-md body-lg"
        type="button"
      >
        Controleer beschikbaarheid
      </button>
      <button
        class="inline-flex items-center justify-center rounded-md text-strong transition-colors duration-200 disabled:opacity-60 disabled:cursor-not-allowed focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 border border-neutral-dark text-neutral-dark hover:bg-neutral-light focus-visible:ring-neutral-dark/40 px-spacing-lg py-spacing-md body-lg"
        type="button"
      >
        Vraag een offerte aan
      </button>
    </div>
    <p>
      Inclusief licht- en geluidspakket op maat.
    </p>
  </div>
</section>
`;
